pkg/plugin/server.go

```diff
@@ -65,22 +78,36 @@ type Server interface {
 
 // server implements Server.
 type server struct {
+	log               *logrus.Logger
+	logLevelFlag      *logging.LevelFlag
+	flagSet           *pflag.FlagSet
 	backupItemAction  *BackupItemActionPlugin
 	blockStore        *BlockStorePlugin
 	objectStore       *ObjectStorePlugin
 	restoreItemAction *RestoreItemActionPlugin
 }
 
 // NewServer returns a new Server
-func NewServer(log logrus.FieldLogger) Server {
+func NewServer() Server {
+	log := newLogger()
+
 	return &server{
+		log:               log,
```

Comment:
Could `newLogger()` be called on this line directly?

Reply:
no, it needs to be declared ahead of time so it can be referenced in the subsequent lines of the struct def